cmd/telliot/main.go

```diff
@@ -110,331 +37,94 @@ func migrateAndOpenDB() (db.DB, error) {
-			}
-
-			cnt := 0
-			start := time.Now()
-			for {
-				cnt++
-				dsStopped := false
-				minerStopped := false
-
-				if ds != nil {
-					dsStopped = !ds.Running
-				} else {
-					dsStopped = true
-				}
-
-				if miner != nil {
-					minerStopped = !miner.Running
-				} else {
-					minerStopped = true
-				}
-
-				if !dsStopped && !minerStopped && cnt > 60 {
-					level.Warn(logger).Log("msg", "taking longer than expected to stop operations", "waited", time.Since(start))
-				} else if dsStopped && minerStopped {
-					break
-				}
-				time.Sleep(500 * time.Millisecond)
-			}
-			level.Info(logger).Log("msg", "main shutdown complete")
+		account, err := rpc.NewAccount(cfg)
+		if err != nil {
+			return nil, nil, nil, errors.Wrap(err, "getting private key to ECDSA")
 		}
-	}
-}
-
-func dataserverCmd(logSetup func(string) log.Logger) func(*cli.Cmd) {
-	return func(cmd *cli.Cmd) {
-		cmd.Action = func() {
-			logger := logSetup(logLevel)
-			// Create os kill sig listener.
-			c := make(chan os.Signal, 1)
-			signal.Notify(c, os.Interrupt)
-
-			var ds *ops.DataServerOps
-			ExitOnError(AddDBToCtx(true), "initializing database")
-			ch := make(chan os.Signal)
-			var err error
-			ds, err = ops.CreateDataServerOps(ctx, logger, config.GetConfig(), database, &proxy, clt, &cont, &acc, ch)
-			ExitOnError(err, "creating data server")
-
-			// Start and wait for it to be ready
-			err = ds.Start(ctx)
-			ExitOnError(err, "starting data server")
-
-			<-ds.Ready()
-
-			http.Handle("/metrics", promhttp.Handler())
-			cfg := config.GetConfig()
-			srv, err := rest.Create(ctx, proxy, cfg.ServerHost, cfg.ServerPort)
-			ExitOnError(err, "creating data server instance")
-			srv.Start()
 
-			// Wait for kill sig.
-			<-c
-			// Notify exit channels.
-			ch <- os.Interrupt
-
-			if err := srv.Stop(); err != nil {
-				level.Warn(logger).Log("msg", "shutting down the server", "err", err)
-			}
-
-			cnt := 0
-			start := time.Now()
-			for {
-				cnt++
-				dsStopped := false
-
-				if ds != nil {
-					dsStopped = !ds.Running
-				} else {
-					dsStopped = true
-				}
-
-				if !dsStopped && cnt > 60 {
-					level.Warn(logger).Log("msg", "taking longer than expected to stop operations", "waited", time.Since(start))
-				} else if dsStopped {
-					break
-				}
-				time.Sleep(500 * time.Millisecond)
-			}
-			level.Info(logger).Log("msg", "main shutdown complete")
+		// Issue #55, halt if client is still syncing with Ethereum network
+		s, err := client.IsSyncing(ctx)
+		if err != nil {
+			return nil, nil, nil, errors.Wrap(err, "determining if Ethereum client is syncing")
+		}
+		if s {
+			return nil, nil, nil, errors.New("ethereum node is still syncing with the network")
 		}
 
+		return client, &contract, &account, nil
 	}
+	// Not sure why we need this case.
+	return nil, nil, nil, nil
+}
+
+var cli struct {
+	Config   configPath  `required type:"existingfile" help:"path to config file"`
```

Comment:
I think to be able to access the config value it needs to be added to every command instead of having it in the root.

Reply:
Look at this example. The arg `Paths` is accessible for the command `RmCmd` through `RmCmd.Paths`

```

type RmCmd struct {
  Force     bool `help:"Force removal."`
  Recursive bool `help:"Recursively remove files."`

  Paths []string `arg name:"path" help:"Paths to remove." type:"path"`
}

func (r *RmCmd) Run(ctx *Context) error {
  fmt.Println("rm", r.Paths)
  return nil
}
var cli struct {
  Rm RmCmd `cmd help:"Remove files."`
}

func main() {
  ctx := kong.Parse(&cli)
  // Call the Run() method of the selected parsed command.
  err := ctx.Run()
  ctx.FatalIfErrorf(err)
}
```